src/subcommand/wallet/inscribe.rs

```diff
@@ -63,7 +65,9 @@ impl Inscribe {
 
     let commit_tx_change = [get_change_address(&client)?, get_change_address(&client)?];
 
-    let reveal_tx_destination = get_change_address(&client)?;
+    let reveal_tx_destination = self
+      .inscription_destination
+      .unwrap_or(get_change_address(&client)?);
```

Comment:
Since `get_change_address` will do a little work, and allocate a change address, you can use `unwrap_or_else`, which takes a closure that is only ever called if the `Option` is `None`:

```suggestion
      .unwrap_or_else(|| get_change_address(&client)?);
```

In general, this should be preferred when calling a function that might be a little slower, or which might have side effects.

This `_else` naming pattern is common, lots of combinators like `Result` have it too. Sometimes it's also called `foo` and `with_foo`, where `foo` takes a value, and `with_foo` takes a closure that returns a value.

Reply:
oh right, `unwrap_or` will eagerly evaluate the default. great catch!

Reply:
So it turns out you can't put the `?` in a closure for `unwrap_or_else`, which makes sense. So I put an `expect` on that instance of `get_change_address`. if you'd prefer expect-free code, I'm happy to change it back